### PR TITLE
Frontend

### PR DESCRIPTION
**Pull request by BojithaPiyathilake**

### CODE REVIEW

File: frontend/package.json [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    "prop-types": "15.6.0",
+    "react": "^16.13.1",
+    "react-animation-components": "3.0.0",
+    "react-dom": "^16.13.1",
+    "react-popper": "0.9.2",
+    "react-redux-form": "1.16.8",
+    "react-scripts": "3.4.1",
+    "react-transition-group": "2.3.0",
+    "reactstrap": "5.0.0",
+    "redux": "3.7.2"
```
Comment (yashodgayashan) [inline frontend/package.json:23]: Why you have added redux?

File: frontend/src/App.css [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    padding: 50px 0px 50px 0px;
+    border-bottom: 1px ridge;
+    min-height:400px;
+}
+
+.footer{
+    background-color: #D1C4E9;
+    margin:0px auto;
+    padding: 20px 0px 20px 0px;
+}
```
Comment (yashodgayashan) [inline frontend/src/App.css:17]:
```suggestion
}

```

File: frontend/src/components/AddModal.js [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+                  className="form-control"
+                  validators={{
+                    required,
+                    minLength: minLength(3),
+                    maxLength: maxLength(20),
+                  }}
+                />
+                <Errors
+                  className="text-danger"
+                  model=".username"
```
Comment (yashodgayashan) [inline frontend/src/components/AddModal.js:91]: is this same error model or different?

File: frontend/src/components/AdminComponent.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+import AddModal from "./AddModal";
+import HeaderComponent from "./HeaderComponent";
+import { userContext } from "../context/userContext";
+
+const AdminComponent = () => {
+  const { users, getUsers, addUser, deleteUser } = useContext(userContext);
+
+  useEffect(() => {
+    getUsers();
+  }, []);
```
Comment (yashodgayashan) [inline frontend/src/components/AdminComponent.js:12]: please check whether this function calling the getUser() continuously.

File: frontend/src/components/EditdetailsModal.js [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+            <Button type="submit" value="submit" color="primary">
+              Submit
+            </Button>
+          </LocalForm>
+        </ModalBody>
+      </Modal>
+    </>
+  );
+}
+export default EditdetailsModal;
```
Comment (yashodgayashan) [inline frontend/src/components/EditdetailsModal.js:182]:
```suggestion
export default EditdetailsModal;

```

File: frontend/src/components/HeaderComponent.js [hunk 6]
```diff
@@ -0,0 +1,37 @@
+import React from "react";
+import { withRouter } from "react-router-dom";
+import { Navbar, NavbarBrand, Nav, NavItem, Button } from "reactstrap";
+
+const HeaderComponent = (props) => {
+  const handleLogOut = (event) => {
+    let confirmlogout = window.confirm("Are you sure you wish to logout?");
+    if (confirmlogout) {
+      alert("User Logged Out");
+      fetch("http://localhost:3000/users/logout")
```
Comment (yashodgayashan) [inline frontend/src/components/HeaderComponent.js:10]: Try to use constant here and then it is easy to change those ports and hosts.

File: frontend/src/components/LoginComponent.js [hunk 7]
```diff
@@ -0,0 +1,174 @@
...
+    type: "admin"
+  });
+  const [userData, setUserData] = useState({});
+
+  const handleSubmit = (values) => {
+    if (login.type === "admin" && login.approved === true){
+      props.history.push("/admin");
+    } 
+    else if (login.type === "student" && login.approved === true){
+
```
Comment (yashodgayashan) [inline frontend/src/components/LoginComponent.js:23]:
```suggestion
```

File: frontend/src/components/LoginComponent.js [hunk 8]
```diff
@@ -0,0 +1,174 @@
...
+      props.history.push("/admin");
+    } 
+    else if (login.type === "student" && login.approved === true){
+      
+      props.history.push("/student");
+    } else{
+      alert('Incorrect Username or Password Entered')
+    }
+
+    /*setLogin({
```
Comment (yashodgayashan) [inline frontend/src/components/LoginComponent.js:29]: is this needed?

File: frontend/src/components/LoginComponent.js [hunk 9]
```diff
@@ -0,0 +1,174 @@
...
+      .then((response) => response.json())
+      .then((item) => {
+        //checkType(item);
+        alert(JSON.stringify(item));
+        console.log(JSON.stringify(item))
+      })
+      .catch((err) => alert(err));
+  };
+
+  /*const checkType = (item) => {
```
Comment (yashodgayashan) [inline frontend/src/components/LoginComponent.js:59]: please remove unnecessary comments

File: frontend/src/components/LoginComponent.js [hunk 10]
```diff
@@ -0,0 +1,174 @@
...
+      .then((response) => response.json())
+      .then((users) => setUserData(users))
+      .catch((err) => console.log(err));
+    }
+  };*/
+
+  const Login = () => {
+    return (
+      <div className="container">
+      <nav class = "navbar navbar-dark bg-primary">
```
Comment (yashodgayashan) [inline frontend/src/components/LoginComponent.js:84]:
```suggestion
      <nav className = "navbar navbar-dark bg-primary">
```

**Review by yashodgayashan — CHANGES_REQUESTED** [review]

File: frontend/package.json (hunk 1 above)
Comment (BojithaPiyathilake) [reply frontend/package.json:23]: That's just to use the redux form components like control. Not for state management.

File: frontend/src/components/AddModal.js (hunk 3 above)
Comment (BojithaPiyathilake) [reply frontend/src/components/AddModal.js:91]: Redux error management for forms.
Every form component will have their own error types. 
Some components have the same types.
The errors are declared at the top.

File: frontend/src/components/AdminComponent.js (hunk 4 above)
Comment (BojithaPiyathilake) [reply frontend/src/components/AdminComponent.js:12]: Yes it worked when i was using the json-server. After integrating with the backend there are problems and when coming upto that point i get an error saying you are not authenticated.

File: frontend/src/components/HeaderComponent.js (hunk 6 above)
Comment (BojithaPiyathilake) [reply frontend/src/components/HeaderComponent.js:10]: ok ayye

File: frontend/src/components/LoginComponent.js (hunk 8 above)
Comment (BojithaPiyathilake) [reply frontend/src/components/LoginComponent.js:29]: Still testing to see if i can get a response from the backend. After a proper response is confirmed the top part can be removed and the commented part can be used as the actual system code.

File: frontend/src/components/LoginComponent.js (hunk 9 above)
Comment (BojithaPiyathilake) [reply frontend/src/components/LoginComponent.js:59]: Even this part will be used once a proper response from the backend is received

File: frontend/src/components/LoginComponent.js (hunk 10 above)
Comment (BojithaPiyathilake) [reply frontend/src/components/LoginComponent.js:84]: Will change

**Pull request merged**